arrow/arrow.py

```diff
@@ -1057,6 +1057,79 @@ def humanize(
                 )
             )
 
+    def dehumanize(self, timestring):
+        """ Returns an arrow object relative to the humanized difference in time given.
+            Only works with specifically formatted strings for the time being.
+
+        :param timestring: A string in the format returned by humanize by default
+            Number must be before unit of time, i.e. (4 hours)
+
+        Usage:
+
+            >>> arrow.utcnow()
+            <Arrow [2020-04-17T00:57:03+00:00]>
+
+            >>> arrow.utcnow().dehumanize("4 hours ago")
+            <Arrow [2020-04-17T20:57:03+00:00]>
+
+            >>> arrow.utcnow().dehumanize("in 4 hours")
+            <Arrow [2020-04-18T04:57:03+00:00]>
+
+            >>> arrow.utcnow().dehumanize("4 days 7 hours 10 minutes 5 seconds ago")
+            <Arrow [2020-04-12T19:46:58+00:00]>
+        """
+
+        current = self.fromdatetime(self._datetime)
+
+        times = timestring.split(" ")
+
+        if times[-1] == "ago":
+            shiftVal = -1
+            times = times[:-1]
+        elif times[0] == "in":
+            shiftVal = 1
+            times = times[1:]
+        else:
+            raise ValueError("Invalid prefix or suffix")
+
+        if len(times) % 2 != 0:
+            raise ValueError("Invalid time input")
+
+        month = 0
+        year = 0
+
+        for i in range(0, len(times), 2):
+            val = int(times[i])
+            unit = times[i + 1]
+
+            if unit in ["second", "seconds"]:
+                current += shiftVal * timedelta(seconds=val)
+            elif unit in ["minute", "minutes"]:
+                current += shiftVal * timedelta(minutes=val)
+            elif unit in ["hour", "hours"]:
+                current += shiftVal * timedelta(hours=val)
+            elif unit in ["day", "days"]:
+                current += shiftVal * timedelta(days=val)
+            elif unit in ["week", "weeks"]:
+                current += shiftVal * timedelta(weeks=val)
+            elif unit in ["month", "months"]:
+                month += shiftVal * val
+            elif unit in ["year", "years"]:
+                year += shiftVal * val
+            else:
+                raise ValueError("Error parsing time string")
+
+        month = current.month + month
+        year += month // 12
+        year = current.year + year
+        month = month % 12
+
+        dateobj = datetime(
+            year, month, current.day, current.hour, current.minute, current.second
+        )
```

Comment:
I think (not fully tested though) that we could use arrow's [shift](https://github.com/crsmithdev/arrow/blob/master/arrow/arrow.py#L634) method here to avoid having to calculate the change manually.